test/integration/product-import.spec.coffee

```diff
@@ -216,6 +236,98 @@ describe 'Product Importer integration tests', ->
       expect(@import._summary.productsWithMissingSKU).toBe(1)
       done()
 
+  it 'should assign New state to products without a state specified', (done) ->
+    productDraft = createProduct()[0]
+
+    @import.performStream([productDraft], -> {})
+    .then =>
+      @fetchProducts(@productType.id)
+    .then (result) =>
+      expect(result.body.results.length).toBe(1)
+      product = result.body.results[0]
+
+      expect(product.state.id).toBe(@newState.id)
+      expect(@import._cache["state"][product.state.id].key).toBe("New")
+      done()
+
+  it 'should assign the appropriate state to products with a state specified', (done) ->
+    productDraft = createProduct()[0]
+
+    productDraft.state = {
+      typeId: "state",
+      id: @extantState.id
+    }
+
+    @import.performStream([productDraft], -> {})
+      .then =>
+        @fetchProducts(@productType.id)
+      .then (result) =>
+        expect(result.body.results.length).toBe(1)
+        product = result.body.results[0]
+
+        expect(product.state.id).toBe(@extantState.id)
+        expect(@import._cache["state"][product.state.id].key).toBe("AlsoNewButIWantToCheckPopulatingWithANonDefaultState")
+        done()
+
+  it 'should process updates for products where the update does not specify the state', (done) ->
+    productDraft = _.deepClone(createProduct()[0])
```

Comment:
why `deepClone`?

Reply:
Just copying what was there already in other tests.